### DISCUSSION TITLE
Can I use navigator features without setting up lsp from navigator ?

### DISCUSSION THREAD

**Discussion by iabdelkareem** [ray-x/navigator.lua#194] [General]
First of all thank you for your contributions 🙏 I'm a big fan of lsp signature and I see navigator really promising. 

Is there a way today to use navigator features while setting up the LSP myself?

I want to take control of configuring the other plugins (e.g., lsp install, lsp config and lsp signature) outside of navigator yet hook the navigator features to it.

Thanks

**Comment by ray-x** [comment]
I am not sure if these works
1) you can enable lsp_installer in navigator, it will skip the lsp setup from navigator
2) set lsp.enable=false in your setup.



You need to add `require('navigator.lspclient.mapping').setup()` so the keymap will work for you. You can overwrite the keymaps if you want to setup this part yourself as well.

```lua
local function on_attach(client, bufnr)
	require("navigator.lspclient.mapping").setup({
		client = client,
		bufnr = bufnr,
		cap = client.server_capabilities,
	})
        ...
       -- or do this:
       vim.cmd('nmap <you_key> lua require('navigator.xxx').func())
end
```
